src/network/serverpackethandler.cpp

```diff
@@ -857,11 +857,21 @@ void Server::handleCommand_PlayerItem(NetworkPacket* pkt)
 		return;
 	}
 
-	u16 item;
+	u16 old_wield_idx, new_wield_idx;
+	ItemStack old_itemstack, new_itemstack;
 
-	*pkt >> item;
+	// Get current wielded item and its idx
+	old_wield_idx = playersao->getWieldIndex();
+	playersao->getWieldedItem(&old_itemstack);
 
-	playersao->getPlayer()->setWieldIndex(item);
+	// Deserialize new wield idx from packet
+	*pkt >> new_wield_idx;
+	playersao->getPlayer()->setWieldIndex(new_wield_idx);
```

Comment:
```suggestion
	player->setWieldIndex(new_wield_idx);
```

Reply:
Ahaha, good catch! :D

Reply:
A couple of other `playersao` calls can be changed to directly calling from `player` too, as the SAO class simply wraps around the respective `Player` methods, which I'm guessing is for convenience.

Reply:
Done